arch/arm/src/armv7-m/arm_tcbinfo.c

```diff
@@ -112,6 +112,8 @@ const struct tcbinfo_s g_tcbinfo =
   TCB_STATE_OFF,
   TCB_PRI_OFF,
   TCB_NAME_OFF,
+  TCB_REGS_OFF,
+  17,
```

Comment:
Don't we have a macro for thus value?  
Also, I see that the `tcbinfo_s` struct has now another member, given the initialization way, is `XCPTCONTEXT_REGS` in the right position here?

Reply:
> Don't we have a macro for thus value?

This value is only used by gcc to get the basic register list which is different from the full list.

> Also, I see that the `tcbinfo_s` struct has now another member, given the initialization way, is `XCPTCONTEXT_REGS` in the right position here?

 Adding this value is avoid to hardcode this information in nuttx-jlink.c:
https://github.com/apache/incubator-nuttx/pull/5966/files#diff-3ad0f365e6bae822bd903a59b51f56f84b12cde99627b13bca8870298e21269dL603-R638

Reply:
I noticed that, but in this case what's the `total_num` member initialized to?

Reply:
It already initialize to XCPTCONTEXT_REGS in code.

Reply:
Quite strange that the designated initializer is used few lines below (`.p =`), but not here.

Reply:
Because the type is union not a pointer in that location.

Reply:
I mean that all other members of `struct tcbinfo_s` can be initialized using designated initializer

Reply:
Done.